src/dependencies.jl

```diff
@@ -18,35 +18,39 @@ end
 
 function get_project_deps(project_file::AbstractString; include_jll::Bool=false)
     project_deps = Set{DepInfo}()
+    dep_section = Dict{DepInfo,String}()
     project = TOML.parsefile(project_file)
 
-    if haskey(project, "deps")
-        deps = project["deps"]
-        add_compat_section!(project)
-        compat = project["compat"]
-
-        for dep in deps
-            name = dep[1]
-            uuid = UUIDs.UUID(dep[2])
-
-            # Ignore JLL packages if flag set
-            # Do NOT ignore stdlib packages.
-            if !endswith(lowercase(strip(name)), "_jll") || include_jll
-                package = Package(name, uuid)
-                compat_entry = DepInfo(package)
-                dep_entry = convert(String, strip(get(compat, name, "")))
-
-                if !isempty(dep_entry)
-                    compat_entry.version_spec = semver_spec(dep_entry)
-                    compat_entry.version_verbatim = dep_entry
+    for section in ["deps", "weakdeps"]
+        if haskey(project, section)
+            deps = project[section]
+            add_compat_section!(project)
+            compat = project["compat"]
+
+            for dep in deps
+                name = dep[1]
+                uuid = UUIDs.UUID(dep[2])
+
+                # Ignore JLL packages if flag set
+                # Do NOT ignore stdlib packages.
+                if !endswith(lowercase(strip(name)), "_jll") || include_jll
+                    package = Package(name, uuid)
+                    compat_entry = DepInfo(package)
+                    dep_entry = convert(String, strip(get(compat, name, "")))
+
+                    if !isempty(dep_entry)
+                        compat_entry.version_spec = semver_spec(dep_entry)
+                        compat_entry.version_verbatim = dep_entry
+                    end
+
+                    push!(project_deps, compat_entry)
+                    get!(dep_section, compat_entry, section)
```

Comment:
what if a package is in multiple sections? E.g. `[deps]` and `[weakdeps]` is a [documented compatibility strategy](https://pkgdocs.julialang.org/v1/creating-packages/#Transition-from-normal-dependency-to-extension)

Reply:
If it’s only the title it seems like not too big of a deal either way, just wanted to make sure semantically it was OK. Maybe a simple choice is to just use deps in the title when present by looping over weak deps then deps (or vice versa if you prefer)

Reply:
Ah, that's currently how this works. `get!(dep_section, compat_entry, section)` ensures that `dep_section[compat_entry]` is only set once to a section for a given package, and if the package was found in the `deps` section, it's already set, so `weakdeps` will be ignored.

Reply:
If possible, I'd like the titles for regular deps to remain the same as the old titles, and for the new title format to only be used for weakdeps.

Reply:
@DilumAluthge this has already been handled in 3e52390